### PR TITLE
Implement dynamic sub-menus

### LINKED ISSUE #207: Implementing dynamic sub-menus
Dynamic sub-menus will allow proper space expansion and allocation, and will greatly aid the toolbar UIs (especially "popup). I'll try to keep the API as much exposed and semantic as possible.

Based off @jywarren's comment [here](https://github.com/publiclab/Leaflet.DistortableImage/pull/155#issuecomment-481886381). Using the [`Leaflet.toolbar`](https://github.com/Leaflet/Leaflet.toolbar) to incorporate this functionality.

/cc @sashadev-sky

### PR DESCRIPTION
**Pull request by rexagod**
Fixes #207. Static implementation. Tests and dynamic implementations will be in a followup PR, as per @jywarren's [suggestion](https://github.com/publiclab/Leaflet.DistortableImage/issues/207#issuecomment-484677133).

/cc @jywarren @sashadev-sky
***
submenu

### CODE REVIEW

File: dist/leaflet.distortableimage.css [hunk 1]
```diff
@@ -40,3 +40,16 @@
 img.leaflet-image-layer.selected {
   box-shadow: 0px 0px 0px 12px #ffea00;
 }
+
```
Comment (jywarren) [inline dist/leaflet.distortableimage.css]: Is this for this PR? Or is it from another, maybe? Thanks!

File: src/edit/DistortableImage.EditToolbar.js [hunk 2]
```diff
@@ -1,162 +1,163 @@
...
-			html: '<span class="fa fa-adjust"></span>',
-			tooltip: 'Toggle Image Transparency',
-			title: 'Toggle Image Transparency'
-		}},
-
-		addHooks: function() {
-			var editing = this._overlay.editing;
-
-			editing._toggleTransparency();
+    initialize: function(map, overlay, customAction, options) {
```
Comment (jywarren) [inline src/edit/DistortableImage.EditToolbar.js]: This looks good, but where is it called from, and how does `customAction` relate to the submenus? Is it related to this PR? I think perhaps we need to see this in a test to better understand how it works; would you mind adding the test to this PR? Thank you!

File: dist/leaflet.distortableimage.css (hunk 1 above)
Comment (rexagod) [reply dist/leaflet.distortableimage.css]: @jywarren, These are for the CSS manipulations for the popup menu I mentioned above.

File: src/edit/DistortableImage.EditToolbar.js (hunk 2 above)
Comment (rexagod) [reply src/edit/DistortableImage.EditToolbar.js]: @jywarren `customAction` offers a deeper degree of control over some actions, which I thought was actually unnecessary for this PR right now, so I've removed it. I'll reintroduce it if we feel the need to, along with the docs.

**Comment by sashadev-sky** [comment]
Noting that I haven't forgotten about this one - but before reviewing this i'd like to address the questions I brought up in https://github.com/publiclab/Leaflet.DistortableImage/pull/169 and maybe get that merged 1st, because it discusses file structure

**Comment by rexagod** [comment]
@sashadev-sky @jywarren I've shared some thoughts on #169, and meanwhile we're discussing there can we progress things here (reviews, etc.) and get this upto speed to a point when only merging is left? I guess that'd be more convenient, and any little adjustments in this PR can be easily made later on. Thanks!

**Comment by jywarren** [comment]
That would be great @rexagod -- i'm eager to get this integrated. Will the rebase here be difficult? Shall we start there, and I'll comment on #169 so we can move here next?

**Comment by rexagod** [comment]
@jywarren Can you let me know which icons do you think should I send to the sub-menu? Also @sashadev-sky can you link me to the svg source so I can fetch an appropriate icon for the sub-menu? Thanks!

image

**Comment by jywarren** [comment]
Great idea. Let's aim for a bulleted list with subitems here as our
discussion phase. Sasha, maybe you first? Or Pranshu if you want to post a
suggested nesting first? I think let's aim to have 4-6 critical tools and
the remainder under a `...` "more tools" menu?

Thanks!

**Comment by rexagod** [comment]
@jywarren I guess an optimal approach here would be to keep the most frequently/important 3 or 4  icons out of the submenu while grouping the other ones. So, I'm guessing,

- Delete
- EXIF
- Revert
- Export

could go inside the submenu. What do you think?

**Comment by jywarren** [comment]
I like it!

**Comment by themacboy** [comment]
Hello guys, 

nice idea and work. I will add one suggestion:

A menu option with sub-menus selectors for each mode, to direct change to desired mode ;)

**Comment by sashadev-sky** [comment]
+1 to the mode idea thank you @themacboy!! I have been trying to figure out ways to clarify that `mode`s are like special actions

**Comment by sashadev-sky** [comment]
@rexagod @jywarren I am thinking that submenus should be for _related_ tools because they imply some sort of categorization / relationship right?, like the modes idea suggested by @themacboy. For extra tools, it would be cool maybe to have them in a little box kinda similar to the way I made this keymapper:


62043675-ef703b80-b1ce-11e9-94f8-b8854446a08e

and you can drag them and replace them with current toolbar icons. It seems difficult but now that we have the code down for making said little box were not that far away from having that right?? Or like to start something as simple as double-clicking them makes them swap out based on LIFO or something

**Comment by jywarren** [comment]
I like the idea of related tools. However, for an initial implementation,
just to get the basic goal of "reduce overwhelming # of tools at first
glance", I think it's OK to have a final menu item of "..." which contains
any tool that's not super critical. Then we can follow up with some other
menu organization tactics!

**Pull request closed without merging**